### PR TITLE
Check all .so's are noexecstack (#3994)

### PR DESCRIPTION
**Pull request by jlaine**

### CODE REVIEW

File: .jenkins/Jenkinsfile-cryptography-wheel-builder [hunk 1]
```diff
@@ -134,6 +134,8 @@ def build(version, label, imageName) {
                         CFLAGS="-I/opt/pyca/cryptography/openssl/include -Wl,--exclude-libs,ALL" \
                         $linux32 /opt/python/$version/bin/pip wheel cryptography==$BUILD_VERSION -w tmpwheelhouse/ --no-binary cryptography --no-deps
                     $linux32 auditwheel repair tmpwheelhouse/cryptography*.whl -w wheelhouse/
+                    unzip wheelhouse/*.whl -d execstack.check
+                    (execstack execstack.check/cryptography/hazmat/bindings/*.so | grep '^X') && exit 1
```
Comment (reaperhulk) [inline .jenkins/Jenkinsfile-cryptography-wheel-builder]: I don't believe execstack is present in the cryptography-manylinux1 image. Additionally, the vagaries of docker in jenkins (which is itself in a docker container) means you'll need to `chmod -R 777 execstack.check` or else we'll have permissions errors on cleanup.

To add execstack you'll want to do a `RUN yum -y install prelink && yum -y clean all` in the existing dockerfile (https://github.com/pyca/infra/blob/master/cryptography-manylinux1/Dockerfile)

Comment (jlaine) [reply .jenkins/Jenkinsfile-cryptography-wheel-builder]: Can I count on "unzip" being present?

Comment (alex) [reply .jenkins/Jenkinsfile-cryptography-wheel-builder]: Yes, unzip is in the image

Comment (reaperhulk) [reply .jenkins/Jenkinsfile-cryptography-wheel-builder]: Okay execstack is now in the container, but the chmod will still need to be done.

**Comment by jlaine** [comment]
For future reference, the output of exectack looks like:

```
- execstack.check/cryptography/hazmat/bindings/_constant_time.abi3.so
- execstack.check/cryptography/hazmat/bindings/_openssl.abi3.so
- execstack.check/cryptography/hazmat/bindings/_padding.abi3.so
```

If I enable exectack on a .so it looks like:

```
- execstack.check/cryptography/hazmat/bindings/_constant_time.abi3.so
X execstack.check/cryptography/hazmat/bindings/_openssl.abi3.so
- execstack.check/cryptography/hazmat/bindings/_padding.abi3.so
```

=> the test is "if any line starts with X", we have a problem

**Pull request merged**